lib/install/management/rbac_test.go

```diff
@@ -62,36 +72,164 @@ func TestRolesVCenter(t *testing.T) {
+}
+
+func TestOpsUserPermsSimulatorVPX(t *testing.T) {
+	ctx := context.Background()
+	m := simulator.VPX()
+
+	defer m.Remove()
+
+	err := m.Create()
+	require.NoError(t, err)
+
+	s := m.Service.NewServer()
+	defer s.Close()
+
+	fmt.Println(s.URL.String())
+
+	config := &session.Config{
+		Service:   s.URL.String(),
+		Insecure:  true,
+		Keepalive: time.Duration(5) * time.Minute,
+	}
+
+	sess, err := session.NewSession(config).Connect(ctx)
+	require.NoError(t, err)
+
+	am := newOpsUserAuthzManager(ctx, sess.Vim25(), nil)
+
+	var roleCount = len(am.targetRoles)
+	count := initRoles(ctx, t, am)
+
+	defer cleanup(ctx, t, am, true)
+	require.Equal(t, roleCount, count, "Incorrect number of roles: expcted %d, actual %d", roleCount, count)
+
+	c := sess.Client
+	// Find the Datacenter
+	finder := find.NewFinder(c.Client, false)
+
+	dcList, err := finder.DatacenterList(ctx, "/*")
+	require.NoError(t, err)
+	require.NotEqual(t, 0, len(dcList))
+
+	dc := dcList[0]
+	finder.SetDatacenter(dc)
+
+	resourcePermission, err := am.addPermission(ctx, dc.Reference(), Datacenter, false)
+	require.NoError(t, err)
+	require.NotNil(t, resourcePermission)
+
+	// Get permission back
+	permissions, err := am.getPermissions(ctx, dc.Reference())
+
+	if err != nil || len(permissions) == 0 {
+		t.Fatalf("Failed to get permissions for Datacenter")
+	}
+
+	foundPermission := permissions[0]
 
-	doTestRoles(ctx, t, mgr)
+	permission := &resourcePermission.permission
+
+	if foundPermission.Principal != permission.Principal ||
+		foundPermission.RoleId != permission.RoleId ||
+		foundPermission.Propagate != permission.Propagate ||
+		foundPermission.Group != permission.Group {
+		t.Fatalf("Permission mismatch, exp: %v, found: %v", permission, foundPermission)
+	}
+}
+
+func TestOpsUserPermsFromConfigSimulatorVPX(t *testing.T) {
+	ctx := context.Background()
+	m := simulator.VPX()
+
+	m.Datacenter = 3
+	m.Folder = 2
+	m.Pool = 1
+	m.App = 1
+	m.Pod = 1
+
+	defer m.Remove()
+
+	err := m.Create()
+	require.NoError(t, err)
+
+	s := m.Service.NewServer()
+	defer s.Close()
+
+	fmt.Println(s.URL.String())
+
+	config := &session.Config{
+		Service:   s.URL.String(),
+		Insecure:  true,
+		Keepalive: time.Duration(5) * time.Minute,
+	}
+	sess, err := session.NewSession(config).Connect(ctx)
+	require.NoError(t, err)
+
+	configSpec, err := validate.TestValidateForSim(ctx, s.URL)
+	require.NoError(t, err)
+
+	// Set up the Authz Manager
+	am := newOpsUserAuthzManager(ctx, sess.Vim25(), configSpec)
+
+	resourcePermissions, err := am.SetupRolesAndPermissions(ctx)
+	require.NoError(t, err)
+	defer cleanup(ctx, t, am, true)
+	require.True(t, len(am.rbacConfig.Resources) >= len(resourcePermissions))
+
+	verifyResourcePermissions(ctx, t, am, resourcePermissions, am.rbacConfig.Resources)
+}
+
+func verifyResourcePermissions(ctx context.Context, t *testing.T, am *AuthzManager, retPerms []RBACResourcePermission, configResources []RBACResource) {
+	// Verify resource permissions
```

Comment:
Redundant comment - this is just the function name. We should either expand on the comment here, or include more details in a function comment above the function.